File: Server/AttributeContainer.dart
import 'dart:convert';

class AttributeContainer{
  Map<String, dynamic> data = new Map<String, dynamic>();
  String key;
  dynamic value;
  AttributeContainer(String key, dynamic value){
    this.key = key;
    this.value = value;
    data[key] = value;
  }

  String toJSON(){
    return json.encode(data);
  }

  String toString(){
    return "{ \"" + key + "\" : " + value.toString() + " }";
  }

}

File: Server/AttributeContainer.go
package main

import (
	"encoding/json"
	"fmt"
)

type AttributeContainer struct {
	Key   string      `json:"key"`
	Value interface{} `json:"value"`
}

//Default Constructor
func (ac *AttributeContainer) New(key string, value interface{}) {
	ac.Key = key
	ac.Value = value
}

func (ac AttributeContainer) ToMap() map[string]interface{} {
	m := make(map[string]interface{})
	m["key"] = ac.Key
	m["value"] = ac.Value
	return m
}

//Converts the container to Json text
func (ac AttributeContainer) ToJson() string {
	js, _ := json.Marshal(ac.ToMap())
	return string(js)
}

//Converts the container to a string
func (ac AttributeContainer) String() string {
	return fmt.Sprintf("{ \" %s \" : %v  }", ac.Key, ac.Value)
}
